src/lib/mathlib/math/Matrix.hpp

```diff
@@ -491,6 +491,49 @@ class __EXPORT Matrix<3, 3> : public MatrixBase<3, 3>
 
 		return euler;
 	}
+
+	/**
+	 * get euler angles from rotation matrix and avoid gimbal lock effects
+	 * euler.data[*] are the elements of the rotation matrix from body to nav frame (NED), i.e. Xn = Tbn * Xb
+	 * This routine checks for the Aircrafts flight regime and rotates the coordinate frame accordingly to avoid gimbal lock effects.
+	 * In plots you will notice a sudden 90° pitch angle change and roll/yaw axis of the body frame switch. Roll/Yaw in the NED framne remain the same.
+	 * coordinate_frame 0 = nose down
```

Comment:
I am not sure if I get you correctly, but this is really only done for displaying in qgc/ log data. Internally the controller runs independently of this function straight on dcm. So we do not change the dcm matrix but rather change the extraction logic so that it displays zero pitch in hover and cruise regime.